src/git/repo.rs

```diff
@@ -608,6 +664,51 @@ Either create it, or update the main branch setting by running:
         }
     }
 
+    /// Returns the current status of the repo index and working copy.
+    pub fn get_status(&self, git_run_info: &GitRunInfo) -> eyre::Result<Vec<StatusEntry>> {
+        let output = git_run_info.run_silent(
+            self,
+            None,
+            &["status", "--porcelain=v2", "--untracked-files=no", "-z"],
+        )?;
+
+        let not_null_terminator = |c: &&u8| **c != 0_u8;
+        let mut statuses = Vec::new();
+        let mut status_bytes = output.iter().peekable();
```

Comment:
Use `output.into_iter()` here so that we can get rid of a layer of references and deal with `&u8` instead of `&&u8`. Then you should also be able to just write `StatusEntry::from_bytes(&line)`.

Reply:
👍